drivers/dma/dma_dw_common.c

```diff
@@ -475,15 +475,17 @@ int dw_dma_start(const struct device *dev, uint32_t channel)
 	dw_write(dev_cfg->base, DW_LLP(channel), llp);
 	LOG_DBG("ctrl_lo %x, masked ctrl_lo %x, LLP %x",
 		lli->ctrl_lo, masked_ctrl_lo, dw_read(dev_cfg->base, DW_LLP(channel)));
-#else
+#endif /* CONFIG_DMA_DW_HW_LLI */
+
+#if !CONFIG_DMA_DW_HW_LLI || CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
 	/* channel needs to start from scratch, so write SAR and DAR */
 	dw_write(dev_cfg->base, DW_SAR(channel), lli->sar);
 	dw_write(dev_cfg->base, DW_DAR(channel), lli->dar);
 
 	/* program CTL_LO and CTL_HI */
 	dw_write(dev_cfg->base, DW_CTRL_LOW(channel), lli->ctrl_lo);
 	dw_write(dev_cfg->base, DW_CTRL_HIGH(channel), lli->ctrl_hi);
```

Comment:
I don't get how this can even work on MTL.
It might be fine to skip writing the SAR/DAR (it is not, really, with how we setup the CTRL_LO) but not writing to CTRL_LO sounds incorrect.
It contains two flags: DW_CTLL_LLP_S_EN and DW_CTLL_LLP_D_EN (both defaults to 0).
The rule to use LLP/LLI transfer is:
LLP_EN is set for the block (design time decision to allow LLP)
LLP_LOC (DW_LLP) register is not 0 - the address of the first LLI is set
DW_CTLL_LLP_S_EN / DW_CTLL_LLP_D_EN is set to 1

If we don't write CTRL_LO, the LLP should not be used, and since we are not writing to SAR/DAR either, we would have a copy from address 0 to address 0?

Or in MTL the DW_CTLL_LLP_x_EN is defaulting to 1?